quickstarts/microsoft.compute/vm-windows-baseline/main.bicep

```diff
@@ -0,0 +1,224 @@
+@description('The Azure region where resources in the template should be deployed.')
+param location string = resourceGroup().location
+
+@description('Name for the subnet in the virtual network where the network interface is connected.')
+param subnetName string = 'subnet'
+
+@description('Name of the network security group that sets rules for the network environment.')
+param networkSecurityGroupName string = 'nsg'
+
+@description('Name of the virtual network.')
+param virtualNetworkName string = 'network'
+
+@description('Network subnet and prefix.')
+param addressPrefix string = '10.2.0.0/16'
+
+@description('Name for the Public IP used to access the Virtual Machine.')
+param publicIpName string = 'myPublicIP'
+
+@description('Allocation method for the Public IP used to access the Virtual Machine.')
+@allowed([
+  'Dynamic'
+  'Static'
+])
+param publicIPAllocationMethod string = 'Static'
+
+@description('SKU for the Public IP used to access the Virtual Machine.')
+@allowed([
+  'Basic'
+  'Standard'
+])
+param publicIpSku string = 'Standard'
+
+@description('Name of the virtual machine.')
+param vmName string = 'hardened-winvm'
+
+@description('The Windows version for the VM. This will pick a fully patched image of this given Windows version.')
+@allowed([
+  '2012-R2-Datacenter'
+  '2016-Datacenter'
+  '2019-Datacenter'
+  '2019-Datacenter-Core'
+  '2019-Datacenter-Core-smalldisk'
+  '2019-Datacenter-smalldisk'
+  '2022-datacenter'
+  '2022-datacenter-azure-edition-smalldisk'
+  '2022-datacenter-core'
+  '2022-datacenter-core-g2'
+  '2022-datacenter-core-smalldisk'
+  '2022-datacenter-core-smalldisk-g2'
+  '2022-datacenter-g2'
+  '2022-datacenter-smalldisk'
+  '2022-datacenter-smalldisk-g2'
+])
+param OSVersion string = '2022-datacenter-azure-edition-smalldisk'
+
+@description('Specifies the storage account type for OS and data disk.')
+@allowed([
+  'Premium_LRS'
+  'StandardSSD_LRS'
+  'Standard_LRS'
+  'UltraSSD_LRS'
+])
+param osDiskStorageAccountType string = 'Standard_LRS'
+
+@description('Size of the virtual machine.')
+param vmSize string = 'Standard_D2s_v3'
+
+@description('Availability zone number.')
+param zone string = '1'
+
+@description('Username for the Virtual Machine.')
+param adminUsername string
+
+@description('Password for the Virtual Machine.')
+@minLength(12)
+@secure()
+param adminPassword string
+
+@description('Enable accelerated networking.')
+param enableAcceleratedNetworking bool = true
+
+resource networkSecurityGroup 'Microsoft.Network/networkSecurityGroups@2021-05-01' = {
```

Comment:
added rules @bmoore-msft

Reply:
@bmoore-msft any task accomplished through means other than random human clicking ;-)

Reply:
Thinking from a QuickStart perspective... I spin up the VM, then what?  It doesn't appear to be running a workload so the only thing I would do next is to login to the VM and see what this hardening is all about?  And if 3389 isn't open how would I see what this thing is supposed to do for me?

Reply:
We should dissuade people from that behavior. It is not OK to log in to servers just to see what's-what. There are great tools for that, from Azure Policy to run command.

It would be interesting to add port 22 for SSH (yes, on Windows) and add a configuration that enables PowerShell transcription. That is one way to have an interactive, rich, diagnostic experience without compromising security and reliability. That could detract from the intent of this quickstart, however.

Reply:
Either sounds fine - as a customer... what should I do after deploying this?  Can we add something to the readme along the lines of "After deploying this template you can..."?

Reply:
Great idea. Pushed a new block of text.